react/features/app-settings/components/AppSettings.native.js

```diff
@@ -36,42 +52,49 @@ class AppSettings extends AbstractAppSettings {
             <Modal
                 animationType = 'slide'
                 onRequestClose = { this._onRequestClose }
-                presentationStyle = 'fullScreen'
-                style = { styles.modal }
+                presentationStyle = 'overFullScreen'
```

Comment:
Why is this needed? According to the docs it's the same as fullScreen, plus transparency, which we are not using: https://facebook.github.io/react-native/docs/modal.html#presentationstyle It also seems to be the default, so can we just remove the prop altogether?

Reply:
It's needed for the underlying view to be re-layout'ed when rotated, so then this screen receives the aspect ration change. It doesn't work with "fullScreen"

Reply:
I commented it out and it worked for me. Looks like `supportedOrientations` is the only required thing. Can you please double-check?

Reply:
Re-checked, the orientation does work, but the screen doesn't get re-layout'd and the paddings will be outdated. Please test it on an iPhone (the most visible it is on an iPhone X). Open the settings screen in portrait, then rotate the device. The header will still have the safety offset (where it shouldn't) and the side won't have it (while it should). Or is it only me?
![image](https://user-images.githubusercontent.com/9974975/34768969-a23d9e32-f5fc-11e7-8139-38b12bbe914b.png)

Reply:
Isn't the aspectRatio a prop? If so, it would trigger a re-render.

Reply:
Yes, bit it doesn't change

Reply:
I tested this and (I guess because this is a Modal) our aspect ratio detector doesn't kick in, and thus there is no new props when we rotate. Sigh.

Reply:
I "solved" it as follows:

```
    componentWillMount() {
        this._orientationListener = DeviceEventEmitter.addListener(
            'namedOrientationDidChange', () => {
                console.log('ORIENTATION CHANGED');
                this.forceUpdate();
            },
        );
    }

    componentWillUnmount() {
        this._orientationListener.remove();
    }
```

I'm thinking we might want to use that for the aspect ratio detector...

Reply:
I leave it to your PR then to fix the detector as we agreed.